### PR TITLE
Soc 979

### PR DESCRIPTION
**Pull request by qaga**
Mercury new login tests:
- successful login
- empty username/password not logging user in
- invalid password 
- wrong username

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Created by Qaga on 2015-06-29.
+ */
+public class Login extends NewTestTemplate {
+
+  private static final String
+      ERROR_MESSAGE =
+      "Hm, we don't recognize these credentials. Please try again or register a new account.";
+
+  @Test()
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java]: add groups to all tests

File: src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  public void validUserCanLogIn() {
+    LoginPage loginPage = new LoginPage(driver).get();
+    loginPage.logUserIn(Configuration.getCredentials().userName10,
+                        Configuration.getCredentials().password10);
+
+    Assertion
+        .assertTrue(loginPage.getNav().isUserLoggedIn(Configuration.getCredentials().userName10));
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java]: this value is used twice in the test:
Configuration.getCredentials().userName10 
move it to a variable

Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java]: Don't use Configuration.getCredentials() in every test
below the ERROR_MESSAGE field add:

Credentials credentials = Configuration.getCredentials();

and refer to credentials variable
See how it's done eg. in:
https://github.com/Wikia/selenium-tests/blob/master/src/test/java/com/wikia/webdriver/testcases/forumtests/ForumEditModeTests.java

File: src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    Assertion.assertTrue(loginPage.isSubmitButtonDisabled());
+  }
+
+  @Test()
+  public void notPossibleToLogInWhenPasswordFieldBlank() throws InterruptedException {
+    LoginPage loginPage = new LoginPage(driver).get();
+    loginPage.logUserIn(Configuration.getCredentials().userName10, "");
+
+    //we must wait some seconds to make sure button is activated
+    Thread.sleep(1000);
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java]: Why do you use sleep ? It is highly discouraged, 
wait for an element instead

File: src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    Assertion.assertEquals(loginPage.getErrorMessage(), ERROR_MESSAGE);
+  }
+
+  @Test()
+  public void notPossibleToLogInWhenUsernameFieldBlank() throws InterruptedException {
+    LoginPage loginPage = new LoginPage(driver).get();
+    loginPage.logUserIn("", Configuration.getCredentials().password10);
+
+    //we must wait some seconds to make sure button is activated
+    Thread.sleep(1000);
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mobile/mercury/Login.java]: Why do you use sleep ? It is highly discouraged, 
wait for an element instead

**Comment by ludwikkazmierczak** [comment]
LGTM

**Comment by justnpT** [comment]
@ludwikkazmierczak please make sure all reviewer comments were introduced or answered before merging because otherwise it may discourage reviewer to do a Code Review

**Pull request merged**